### PR TITLE
Dev 148/add mirror mode to settings

### PR DESCRIPTION
**Pull request by MiladNazeri**
https://highfidelity.atlassian.net/browse/DEV-631 and https://highfidelity.atlassian.net/browse/DEV-148

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21518/HighFidelity-Beta-PR16247-39d96b7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21518/HighFidelity-Quest-Beta-PR16247-39d96b7.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21518/HighFidelity-Beta-PR16247-39d96b7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21518/HighFidelity-Beta-Interface-PR16247-39d96b7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21518/HighFidelity-Beta-PR16247-39d96b7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21518/HighFidelity-Beta-Interface-PR16247-39d96b7.exe

File: interface/resources/qml/hifi/simplifiedUI/settingsApp/general/General.qml [hunk 1]
```diff
@@ -239,6 +239,16 @@ Flickable {
                         Camera.mode = "third person"
```
Comment (luiscuenca) [inline interface/resources/qml/hifi/simplifiedUI/settingsApp/general/General.qml]: We also need to update the "third person" camera mode to the new "look at" mode:
`Camera.mode = "look at"`
The menu should be still labeled "Third Person View"

File: interface/resources/qml/hifi/simplifiedUI/settingsApp/general/General.qml [hunk 2]
```diff
@@ -239,6 +239,16 @@ Flickable {
...
                     }
                 }
+
+                SimplifiedControls.RadioButton {
+                    id: selfie
+                    text: "Selfie"
+                    checked: Camera.mode === "mirror"
```
Comment (luiscuenca) [inline interface/resources/qml/hifi/simplifiedUI/settingsApp/general/General.qml]: Also update "mirror" to the new mode "selfie"

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21524/HighFidelity-Beta-PR16247-aba6a28.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21524/HighFidelity-Quest-Beta-PR16247-aba6a28.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21524/HighFidelity-Beta-PR16247-aba6a28.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21524/HighFidelity-Beta-Interface-PR16247-aba6a28.dmg

File: interface/resources/qml/hifi/simplifiedUI/helpApp/controls/ControlsTable.qml [hunk 3]
```diff
@@ -699,8 +699,8 @@ Item {
                 spacing: controlsTableRoot.rowPadding
 
                 HifiStylesUit.GraphikRegular {
-                    id: mirrorText
-                    text: "Mirror Mode"
+                    id: selfieText
+                    text: "Selfie"
```
Comment (RebeccaStankus) [inline interface/resources/qml/hifi/simplifiedUI/helpApp/controls/ControlsTable.qml]: Should this also be changed to "Look at self"?

File: interface/resources/qml/hifi/simplifiedUI/settingsApp/general/General.qml [hunk 4]
```diff
@@ -234,9 +234,19 @@ Flickable {
...
-                    checked: Camera.mode === "third person"
+                    checked: Camera.mode === "look at"
                     onClicked: {
-                        Camera.mode = "third person"
+                        Camera.mode = "look at"
+                    }
+                }
+
+                SimplifiedControls.RadioButton {
+                    id: selfie
```
Comment (RebeccaStankus) [inline interface/resources/qml/hifi/simplifiedUI/settingsApp/general/General.qml]: same here...should it be changed to "Look at self"?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21524/HighFidelity-Beta-PR16247-aba6a28.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21524/HighFidelity-Beta-Interface-PR16247-aba6a28.exe

File: interface/resources/qml/hifi/simplifiedUI/helpApp/controls/ControlsTable.qml (hunk 3 above)
Comment (MiladNazeri) [reply interface/resources/qml/hifi/simplifiedUI/helpApp/controls/ControlsTable.qml]: Look at self was just the extra explanation.  The actual setting is "selfie".

New Camera modes btw:
CameraMode stringToMode(const QString& mode) {
    if (mode == "third person") {
        return CAMERA_MODE_THIRD_PERSON;
    } else if (mode == "first person") {
        return CAMERA_MODE_FIRST_PERSON;
    } else if (mode == "mirror") {
        return CAMERA_MODE_MIRROR;
    } else if (mode == "independent") {
        return CAMERA_MODE_INDEPENDENT;
    } else if (mode == "entity") {
        return CAMERA_MODE_ENTITY;
    } else if (mode == "look at") {
        return CAMERA_MODE_LOOK_AT;
    } else if (mode == "selfie") {
        return CAMERA_MODE_SELFIE;
    }
    return CAMERA_MODE_NULL;
}

QString modeToString(CameraMode mode) {
    if (mode == CAMERA_MODE_THIRD_PERSON) {
        return "third person";
    } else if (mode == CAMERA_MODE_FIRST_PERSON) {
        return "first person";
    } else if (mode == CAMERA_MODE_MIRROR) {
        return "mirror";
    } else if (mode == CAMERA_MODE_INDEPENDENT) {
        return "independent";
    } else if (mode == CAMERA_MODE_ENTITY) {
        return "entity";
    } else if (mode == CAMERA_MODE_LOOK_AT) {
        return "look at";
    } else if (mode == CAMERA_MODE_SELFIE) {
        return "selfie";
    }
    return "unknown";
}

**Pull request merged**